src/pages/_app.js

```diff
@@ -34,6 +40,31 @@ function App({ Component, pageProps }) {
     }
   }, []);
 
+  useEffect(() => {
+    // This effect is called everytime the user changes because firebase has updated the token
+    // or if the router changes (i.e the user navigates)
+    const checkTermsAcceptance = async (u) => {
+      const serverUser = await axiosGet('/profile',
+        {
+          headers: {
+            authorization: `Bearer ${u.xa}`,
+          },
+        });
+      return serverUser.termsAccepted;
+    };
+    if (user && user.emailVerified) {
+      checkTermsAcceptance(user).then((termsAccepted) => {
+        // If there is a verified user but they haven't accepted the terms, reroute them to sign-in
+        if (!termsAccepted) {
+          router.push('/sign-in');
+        }
+      });
+    } else if (user && !user.emailVerified) {
+      // If there is a user but they aren't verified log them out
+      firebase.auth().signOut();
```

Comment:
Does this need to redirect to a page where user's are informed as to why they've been signed out?

Reply:
There shouldn't be a case where the user is logged in & they aren't verified as this is also handled by the signin page. However I put in in here as a double check.

Liklihood is if this check gets hit, a user won't even realise they were logged in, so this should all happen seamlessly in the background for them,

Reply:
Ah, fair enough.